hedera-node/docs/design/services/smart-contract-service/scheduled-create-token-system-contract.md

```diff
@@ -0,0 +1,130 @@
+
+### System Contract Module
+
+- `ScheduleNativeTranslator` - This class will be responsible for handling the `scheduleCreateFungibleToken`, `scheduleCreateFungibleTokenWithCustomFees`, `scheduleCreateNonFungibleToken`, `scheduleCreateNonFungibleTokenWithCustomFees`, `scheduleUpdateTokenInfo` and selectors and dispatching them to the Schedule Service.
+- `ScheduleNativeDecoder` - This class provides methods and constants for decoding the given `HssCallAttempt` into the appropriate `TransactionBody` by using the `asDispatchIn` method of the `Call` interfaced described below.
+- `GetScheduledInfoTranslator` - This class will be responsible for handling the `getScheduledFungibleTokenCreateTransaction`, `getScheduledNonFungibleTokenCreateTransaction` and `getScheduledTokenAddress` selectors and dispatching them to the Schedule Service.
+- `GetScheduledTokenInfoCall` - This class provides methods and constants for decoding the `ScheduleGetInfoResponse` into a `PricedResult`.
+- `GetScheduledNonFungibleTokenInfoCall` - This class provides methods and constants for decoding the `ScheduleGetInfoResponse` into a `PricedResult`.
+
+### Supported Function Calls
+In order to maintain control over which functions are supported by the `scheduleNative` function, the call data will be validated to ensure that the function selector matches the following function definitions:
+`createFungibleToken(HederaToken memory token,int64 initialTotalSupply, int32 decimals)`
+`createFungibleTokenWithCustomFees(HederaToken memory token, int64 initialTotalSupply, int32 decimals, FixedFee[] memory fixedFees, FractionalFee[] memory fractionalFees)`
+`createNonFungibleToken(HederaToken memory token)`
+`createNonFungibleTokenWithCustomFees(HederaToken memory token, FixedFee[] memory fixedFees, RoyaltyFee[] memory royaltyFees)`
+`updateTokenInfo(address token, HederaToken memory tokenInfo)`
+
+Call data that does not match these function definitions will result in a `SCHEDULED_TRANSACTION_NOT_IN_WHITELIST` response code.
+
+### New method in Call Interface
+The current `Call` interface will be extended to include a new method `asDispatchIn` that will be used to convert the `Call` object into a `TransactionBody` object.
+Existing functions that produce a `TransactionBody` object will be updated to use this new method.
+
+  ```
+  /**
+  * Encapsulates a call to the HTS system contract.
+  */
+  public interface Call {
+  ...
+    /**
+    * @param frame the message frame
+      * @return the native TransactionBody implied by this call
+      */
+        @NonNull
+        default TransactionBody asDispatchIn(MessageFrame frame) {
+            throw new UnsupportedOperationException("Needs scheduleNative() support");
+        }
+      }
+  ```
+
+### Gas Costs
+
+The gas costs will consist of the price for the call to the Schedule Service plus the pricing for the transaction that would be executed (for creation transactions this would include sending
```

Comment:
I believe we no longer need to charge the price of the transaction to be scheduled as the payer noted in the schedule will do that

Reply:
Adjusted to match the current state